.github/workflows/ci.yml

```diff
@@ -61,11 +61,38 @@ jobs:
       - run: poetry install --no-root --all-extras
       - run: poe lint
       - run: poe build-develop
+
+      - name: Monitor creation of zombie processes
```

Comment:
(note if this is just a temp PR, none of what I said applies)

Reply:
> If this is a check we want to have in tests, all of this should be done in Python code IMO as part of the tests and should be validated across platforms instead of just Unix and instead of using YAML and bash.

I personally did not intend to keep the zombie checks, and checking using `ps` + `grep` was easier for me than implementing equivalent checks in all languages. However, I wanted to make sure that it was working properly and that it was in a form that can be easily and quickly brought back at some later point if we again get reports of zombie processes.

> Also, do you know if there is some test on any repo that guarantees the server shutdown destroys the process? (ideally tested on all platforms)

We don't AFAIK, and I'm pretty sure there are more ways that ephemeral server process handling is currently broken (mostly rare race conditions). It's possible that some of these might be the cause of the "Failed to connect to ephemeral server within 5 seconds" issue observed regularly in TS SDK. At this point, however, I really need to move on, and I don't believe any of this deserves high priority at this moment, so I'll just open tickets in Core + all core based SDKs listing the scenarios I have in mind.

Reply:
I would rather not have platform specific scripts/bash like this in the repo. If we don't have time to properly test the process is gone upon shutdown we can make an issue for it.

I think you can move on. Let's consider this acceptably tested and close the PR without merging and we can add an issue to track getting cross-platform tests to confirm dev server process death upon shutdown.